v2/metadata/client.go

```diff
@@ -325,6 +329,29 @@ func (c *Client) GetExecutions(ctx context.Context, ids []int64) ([]*pb.Executio
 	return res.Executions, nil
 }
 
+// GetEventsByArtifactIDs ...
+func (c *Client) GetEventsByArtifactIDs (ctx context.Context, artifactIds []int64) ([]*pb.Event, error) {
+	req := &pb.GetEventsByArtifactIDsRequest{ArtifactIds: artifactIds}
+	res, err := c.svc.GetEventsByArtifactIDs(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return res.Events, nil
+}
+
+func (c* Client) GetArtifactName(ctx context.Context, artifactId int64)(string, error) {
+	mlmdEvents, err := c.GetEventsByArtifactIDs(ctx, []int64{artifactId})
```

Comment:
Perf: this would create one request per artifact.
When we get the artifacts initially, we already got those events once, can we keep using them to get their names?

Reply:
I think MLMD should be able to handle the load.
I think currently the logic is more clear.
If later we see performance issues, I think we can change it later.

Reply:
One of the major community concerns in bit.ly/kfp-v2 was whether using MLMD can handle large volumes of pipelines. MLMD can easily become a bottleneck.

I think you'll be able to write clear code (probably even clearer) without compromising perf by changing the `GetOutputArtifactsByExecutionId` method signature
https://github.com/kubeflow/pipelines/blob/d7edb607aa178da648a2e4ba6facb76eb701d0b9/v2/metadata/client.go#L369

Instead of getting a list of artifacts (which lacks the artifact name info), we'd want to return map[string]ArtifactList (to also account for there can be multiple artifacts in the same output key).

Reply:
I will create another PR to refactor this method

Reply:
Why will we have  multiple artifacts in the same output key? I thought currently we only have one artifact per one output key(output key will be the artifact name). I know in the future, we may have multiple artifacts in the same output key